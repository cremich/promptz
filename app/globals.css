pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
