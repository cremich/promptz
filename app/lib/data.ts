import {
  PromptCategory,
  QInterface,
  SdlcActivity,
} from "@/app/lib/definitions";

function convertEnumToTags(enumObj: Record<string, string>): string[] {
  return Object.entries(enumObj)
    .filter(([key]) => isNaN(Number(key)))
    .map(([_, value]) => value);
}

export function getAllTags(): string[] {
  const sdlcTags = convertEnumToTags(SdlcActivity);
  const categoryTags = convertEnumToTags(PromptCategory);
  const interfaceTags = convertEnumToTags(QInterface);
  return [...sdlcTags, ...categoryTags, ...interfaceTags];
}

export function getQInterfaceTags(): string[] {
  return convertEnumToTags(QInterface);
}

export function getCategoryTags(): string[] {
  return convertEnumToTags(PromptCategory);
}

export function getSdlcTags(): string[] {
  return convertEnumToTags(SdlcActivity);
}
