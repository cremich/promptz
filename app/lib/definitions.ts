import { z } from "zod";

export enum SdlcActivity {
  DEBUG = "Debugging",
  DEPLOY = "Deploy",
  DESIGN = "Design",
  DOCUMENT = "Documentation",
  ENHANCE = "Enhance",
  IMPLEMENT = "Implement",
  OPERATE = "Operate",
  OPTIMIZE = "Optimize",
  PATCH = "Patch Management",
  PLAN = "Plan",
  REFACTOR = "Refactoring",
  REQ = "Requirements",
  SECURITY = "Security",
  SUPPORT = "Support",
  TEST = "Test",
}

export enum PromptCategory {
  CHAT = "Chat",
  DEV_AGENT = "Dev Agent",
  DOC_AGENT = "Doc Agent",
  INLINE = "Inline",
  REVIEW_AGENT = "Review Agent",
  TEST_AGENT = "Test Agent",
  TRANSFORM = "Transform Agent",
  TRANSLATE = "Translate",
}

export enum QInterface {
  IDE = "IDE",
  CLI = "CLI",
  CONSOLE = "Management Console",
}

export type User = {
  id: string;
  username: string;
  displayName: string;
  guest: boolean;
};

export type Prompt = {
  id?: string;
  title?: string;
  description?: string;
  tags?: string[];
  instruction?: string;
  howto?: string;
  author?: string;
  authorId?: string;
  public?: boolean;
  slug?: string;
  sourceURL?: string;
  createdAt?: string;
  updatedAt?: string;
};

const ALLOWED_DOMAINS = [
  "github.com",
  "dev.to",
  "hashnode.com",
  "medium.com",
  "stackoverflow.com",
  "community.aws",
  "linkedin.com",
  "aws.amazon.com",
  "docs.aws.amazon.com",
  "amazon.com",
  "amazon.science",
  "huggingface.co",
  "kaggle.com",
  "paperswithcode.com",
  "readthedocs.io",
  "gitbook.io",
];

export const promptFormSchema = z.object({
  id: z.string().uuid().optional(),
  sourceURL: z
    .string()
    .max(2048, "URL must not exceed 2048 characters")
    .url()
    .regex(/^https:\/\/.+/, "Only HTTPS URLs are allowed")
    .refine((url) => {
      try {
        const domain = new URL(url).hostname;
        return ALLOWED_DOMAINS.some(
          (allowed) => domain === allowed || domain.endsWith(`.${allowed}`),
        );
      } catch {
        return false;
      }
    }, "Domain not allowed. Please use a supported content platform.")
    .optional()
    .or(z.literal("")),
  title: z
    .string()
    .trim()
    .max(100, "Title must be less than 100 characters")
    .min(3, "Title must be more than 3 characters"),
  description: z
    .string()
    .trim()
    .min(10, "Description must be more than 10 characters")
    .max(500, "Description must be less than 500 characters"),
  howto: z
    .string()
    .trim()
    .max(4000, "How to must be less than 4000 characters")
    .optional(),
  instruction: z
    .string()
    .trim()
    .min(10, "Prompt must be more than 10 characters")
    .max(4000, "Prompt must be less than 4000 characters"),
  tags: z.array(z.string()).optional(),
  public: z.boolean(),
});

// Validation schema for search and filter params
export const searchParamsSchema = z.object({
  query: z.string().optional(),
  sort: z.string().optional(),
  my: z.string().optional(),
  interface: z.union([z.string(), z.array(z.string())]).optional(),
  category: z.union([z.string(), z.array(z.string())]).optional(),
  sdlc: z.union([z.string(), z.array(z.string())]).optional(),
});
