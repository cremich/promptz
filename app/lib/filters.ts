export type FilterCondition = Record<string, unknown>;

// Utility function for case variations
export function getCaseVariations(text: string): string[] {
  return [
    text.toLowerCase(),
    text.toUpperCase(),
    text.charAt(0).toUpperCase() + text.slice(1).toLowerCase(),
  ];
}

// Handle text search conditions
export function buildTextSearchFilter(query: string): FilterCondition {
  const variations = getCaseVariations(query);
  return {
    or: variations.flatMap((variant) => [
      { name: { contains: variant } },
      { description: { contains: variant } },
    ]),
  };
}

// Handle tag-based filtering
export function buildTagFilter(params: string | string[]): FilterCondition[] {
  const tags = Array.isArray(params) ? params : [params];
  return tags.map((tag) => ({
    tags: { contains: tag },
  }));
}
