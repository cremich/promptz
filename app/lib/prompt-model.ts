import {
  idSchema,
  sourceURLSchema,
  titleSchema,
  descriptionSchema,
  tagSchema,
  publicSchema,
} from "@/lib/forms/schema-definitions";
import { z } from "zod";

export type Prompt = {
  id?: string;
  title?: string;
  description?: string;
  tags?: string[];
  instruction?: string;
  howto?: string;
  author?: string;
  authorId?: string;
  public?: boolean;
  slug?: string;
  sourceURL?: string;
  createdAt?: string;
  updatedAt?: string;
  copyCount?: number;
  starCount?: number;
  popularityScore?: number;
};
export const promptFormSchema = z.object({
  id: idSchema,
  sourceURL: sourceURLSchema,
  title: titleSchema,
  description: descriptionSchema,
  howto: z
    .string()
    .trim()
    .max(4000, "How to must be less than 4000 characters")
    .optional(),
  instruction: z
    .string()
    .trim()
    .min(10, "Prompt must be more than 10 characters")
    .max(4000, "Prompt must be less than 4000 characters"),
  tags: tagSchema,
  public: publicSchema,
}); // Validation schema for search and filter params

export const promptSearchParamsSchema = z.object({
  query: z.string().optional(),
  sort: z.string().optional(),
  tags: z.union([z.string(), z.array(z.string())]).optional(),
});
