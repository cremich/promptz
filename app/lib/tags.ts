import {
  PromptCategory,
  QInterface,
  SdlcActivity,
} from "@/app/lib/definitions";

function convertEnumToTags(enumObj: Record<string, string>): string[] {
  return Object.values(enumObj).filter((value) => typeof value === "string");
}

export function getQInterfaceTags(): string[] {
  return convertEnumToTags(QInterface);
}

export function getCategoryTags(): string[] {
  return convertEnumToTags(PromptCategory);
}

export function getSdlcTags(): string[] {
  return convertEnumToTags(SdlcActivity);
}
