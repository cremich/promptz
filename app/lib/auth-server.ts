import { cookies } from "next/headers";
import { NextRequest, NextResponse } from "next/server";
import { createServerRunner } from "@aws-amplify/adapter-nextjs";
import { fetchAuthSession, fetchUserAttributes } from "aws-amplify/auth/server";

import outputs from "@/amplify_outputs.json";
import { User } from "@/app/lib/definitions";

const { runWithAmplifyServerContext } = createServerRunner({
  config: outputs,
});

export async function fetchCurrentAuthUser(): Promise<User> {
  try {
    const currentUser = await runWithAmplifyServerContext({
      nextServerContext: { cookies },
      operation: (contextSpec) => fetchUserAttributes(contextSpec),
    });
    return { displayName: currentUser.preferred_username!, guest: false };
  } catch (error) {
    return { displayName: "", guest: true };
  }
}

export async function checkUserIsAuthenticated(
  request: NextRequest,
  response: NextResponse,
) {
  const authenticated = await runWithAmplifyServerContext({
    nextServerContext: { request, response },
    operation: async (contextSpec) => {
      try {
        const session = await fetchAuthSession(contextSpec, {});
        return session.tokens !== undefined;
      } catch {
        return false;
      }
    },
  });
  return authenticated;
}
