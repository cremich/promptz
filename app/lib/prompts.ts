// utils/amplify-utils.ts
import { cookies } from "next/headers";

import { createServerRunner } from "@aws-amplify/adapter-nextjs";
import { generateServerClientUsingCookies } from "@aws-amplify/adapter-nextjs/api";

import { type Schema } from "@/amplify/data/resource";
import outputs from "@/amplify_outputs.json";
import { Prompt } from "@/app/lib/definitions";

export const { runWithAmplifyServerContext } = createServerRunner({
  config: outputs,
});

export const appsync = generateServerClientUsingCookies<Schema>({
  config: outputs,
  cookies,
});

export async function fetchFeaturedPrompts(): Promise<Prompt[]> {
  console.log("fetched from server");
  const { data: prompts, errors } = await appsync.models.prompt.list({
    limit: 3,
  });

  if (errors && errors.length > 0) {
    throw new Error(errors[0].message);
  }

  const promptList = prompts.map((p) => {
    const tags = Object.entries({
      category: p.category,
      sdlc_phase: p.sdlc_phase,
      interface: p.interface,
    })
      .map(([_, value]) => value)
      .filter(Boolean);

    return {
      id: p.id,
      title: p.name,
      description: p.description,
      tags: tags,
      author: p.owner_username,
    } as Prompt;
  });
  return promptList;
}
