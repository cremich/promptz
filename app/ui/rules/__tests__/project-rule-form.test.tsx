import { ProjectRule } from "@/app/lib/definitions";
import ProjectRuleForm from "@/app/ui/rules/project-rule-form";
import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("Project Rule Form", () => {
  const projectRule: ProjectRule = {
    id: "12345",
    title: "Test Project Rule",
    description: "This is a test project rule",
    content: "# Project Rule Content\nThis is the content of the project rule.",
    tags: ["NextJS", "TypeScript"],
    author: "cremich",
    authorId: "12345",
    public: true,
    sourceURL: "https://github.com/example/repo",
  };

  test("renders the project rule form component unchanged", () => {
    const { container } = render(<ProjectRuleForm projectRule={projectRule} />);
    expect(container).toMatchSnapshot();
  });
});
