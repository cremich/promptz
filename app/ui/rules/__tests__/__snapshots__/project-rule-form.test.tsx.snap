// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Rule Form renders the project rule form component unchanged 1`] = `
<div>
  <form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
    class="space-y-8"
  >
    <input
      name="id"
      type="hidden"
      value="12345"
    />
    <div
      class="grid grid-cols-1 md:grid-cols-[1fr_2fr] gap-4"
    >
      <div>
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm h-full"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
            >
              <svg
                class="lucide lucide-file-text w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
                />
                <path
                  d="M14 2v4a2 2 0 0 0 2 2h4"
                />
                <path
                  d="M10 9H8"
                />
                <path
                  d="M16 13H8"
                />
                <path
                  d="M16 17H8"
                />
              </svg>
              Basic Information
            </div>
            <div
              class="text-sm text-muted-foreground"
            >
              What is this project rule for? What is its purpose?
            </div>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-center gap-2"
                for=":r0:-form-item"
              >
                Title
              </label>
              <input
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-white placeholder-opacity-50"
                id=":r0:-form-item"
                name="title"
                placeholder="A clear and descriptive title for your project rule"
                value="Test Project Rule"
              />
            </div>
            <div
              class="space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-center gap-2"
                for=":r1:-form-item"
              >
                Description
              </label>
              <textarea
                aria-describedby=":r1:-form-item-description"
                aria-invalid="false"
                class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[140px] text-white placeholder-opacity-50"
                id=":r1:-form-item"
                name="description"
                placeholder="What is this project rule for? What problem does it solve?"
              >
                This is a test project rule
              </textarea>
            </div>
            <div
              class="space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-center gap-2"
                for=":r2:-form-item"
              >
                Tags
              </label>
              <div
                aria-describedby=":r2:-form-item-description"
                aria-invalid="false"
                class="flex flex-wrap gap-2"
                id=":r2:-form-item"
              >
                <input
                  name="tags"
                  type="hidden"
                  value="NextJS"
                />
                <input
                  name="tags"
                  type="hidden"
                  value="TypeScript"
                />
                <div
                  class="flex flex-wrap gap-2"
                >
                  <div
                    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent text-secondary-foreground bg-violet-500 hover:bg-violet-500"
                  >
                    NextJS
                  </div>
                  <div
                    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent text-secondary-foreground bg-violet-500 hover:bg-violet-500"
                  >
                    TypeScript
                  </div>
                </div>
                <button
                  aria-controls="radix-:r3:"
                  aria-expanded="false"
                  aria-haspopup="dialog"
                  data-state="closed"
                  type="button"
                >
                  <div
                    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-secondary-foreground bg-neutral-600 border-dashed border-white hover:bg-neutral-600 cursor-pointer"
                  >
                    Edit Tags
                  </div>
                </button>
              </div>
            </div>
            <div
              class="space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 flex items-center gap-2"
                for=":r6:-form-item"
              >
                Source URL
              </label>
              <input
                aria-describedby=":r6:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm text-white placeholder-opacity-50"
                id=":r6:-form-item"
                name="sourceURL"
                placeholder="Optional: Link to original source or inspiration"
                value="https://github.com/example/repo"
              />
              <p
                class="text-sm text-muted-foreground"
                id=":r6:-form-item-description"
              >
                If this project rule was inspired by or adapted from another source, provide the URL here
              </p>
            </div>
            <div
              class="space-y-2 flex flex-row items-center justify-between"
            >
              <div
                class="space-y-0.5"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r7:-form-item"
                >
                  Visibility
                </label>
                <p
                  class="text-sm text-muted-foreground pr-10"
                  id=":r7:-form-item-description"
                >
                  Keep your project rule private as a draft or just for you. A private project rule can still be shared via URL but will not be listed on promptz.dev. Make your project rule public to share it with the community.
                </p>
              </div>
              <div
                aria-describedby=":r7:-form-item-description"
                aria-invalid="false"
                id=":r7:-form-item"
              >
                <button
                  aria-checked="true"
                  class="peer inline-flex h-5 w-9 shrink-0 cursor-pointer items-center rounded-full border-2 shadow-sm transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input border-neutral-400"
                  data-state="checked"
                  role="switch"
                  type="button"
                  value="on"
                >
                  <span
                    class="pointer-events-none block h-4 w-4 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-4 data-[state=unchecked]:translate-x-0"
                    data-state="checked"
                  />
                </button>
                <input
                  aria-hidden="true"
                  checked=""
                  style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
                  tabindex="-1"
                  type="checkbox"
                  value="on"
                />
                <input
                  name="public"
                  type="hidden"
                  value="true"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm h-full"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
            >
              <svg
                class="lucide lucide-terminal w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="4 17 10 11 4 5"
                />
                <line
                  x1="12"
                  x2="20"
                  y1="19"
                  y2="19"
                />
              </svg>
              Rule Content
            </div>
            <div
              class="text-sm text-muted-foreground"
            >
              Enter the content of your project rule in markdown format
            </div>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="space-y-2"
            >
              <textarea
                aria-describedby=":r8:-form-item-description"
                aria-invalid="false"
                class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[500px] text-white placeholder-opacity-50 font-mono"
                id=":r8:-form-item"
                name="content"
                placeholder="# Your Project Rule Content in Markdown  Write your project rule content here using markdown formatting.  Include code examples, guidelines, and best practices."
              >
                # Project Rule Content
This is the content of the project rule.
              </textarea>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center gap-4"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
        type="submit"
      >
        <svg
          class="lucide lucide-save w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.2 3a2 2 0 0 1 1.4.6l3.8 3.8a2 2 0 0 1 .6 1.4V19a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2z"
          />
          <path
            d="M17 21v-7a1 1 0 0 0-1-1H8a1 1 0 0 0-1 1v7"
          />
          <path
            d="M7 3v4a1 1 0 0 0 1 1h7"
          />
        </svg>
        Save Project Rule
      </button>
      <button
        aria-controls="radix-:r9:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 px-4 py-2 ml-auto"
        data-state="closed"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <svg
            class="lucide lucide-trash2 w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 6h18"
            />
            <path
              d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
            />
            <path
              d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
          Delete Project Rule
        </div>
      </button>
    </div>
  </form>
</div>
`;
